docs/source/caching/cache-key-resolution.mdx

```diff
@@ -87,3 +87,161 @@ This is possible if:
+The `@fieldPolicy` directive allows you to specify how cache keys should be resolved when querying a specific field. You can apply the `@fieldPolicy` directive to `Object` and `Interface` types in your schema by extending the type in a `.graphqls` schema extension file. You can also programmatically provide cache keys by implementing the `FieldPolicyProvider` protocol on your existing `SchemaConfiguration`.
+
+The field policy feature is meant to be used in unison with type policies, meaning if you are using either the directive or programmatic field policy implementations you should be doing the same for type policy.
+
+When executing an operation the library will first check programmatic cache keys, then for directive based cache keys, and then fallback to the default cache key implementation, `Protocol > Directive > Default`.
+
+
+### Directive Usage
+
+The `@fieldPolicy` directive has a `forField` and `keyArgs` argument to provide the name of the field on the type that you are adding the field policy to. As well as which input arguments of that field should be used to resolve cache keys, and in what order they should be used in the case of multiple key arguments. The format of the resolved cache key is the same with `@typePolicy` and looks like `{type name}:{value}`, when multiple key arguments are present they are joined with a `+` separator.
+
+Given a type like this:
+```graphql
+type Query {
+  allAnimals(species: String!): Animal!
+}
+
+type Animal {
+  id: ID!
+  name: String!
+  species: String!
+}
+```
+
+You would apply this field policy like this:
+```graphql
+extend type Query @fieldPolicy(forField: "allAnimals", keyArgs: "species")
+```
+
+This results in resolved cache keys that look like `Animal:Lion` or `Animal:Tiger`.
+
+#### Multiple Key Arguments
+
+When you have multiple key arguments, the are provided to the directive with a space delimiter. So when you have a type like this:
+```graphql
+type Query {
+  allAnimals(species: String!, habitat: String!): Animal!
+}
+
+type Animal {
+  id: ID!
+  name: String!
+  species: String!
+  habitat: String!
+}
+```
+
+You would apply this field policy like this:
+```graphql
+extend type Query @fieldPolicy(forField: "allAnimals", keyArgs: "species habitat")
+```
+
+This results in resolved cache keys that look like `Animal:Lion+Savanna`.
+
+You can control the order of arguments in the cache key with the order you provide them in the directive, so this would result 
+in a cache key that looks like `Animal:Savanna+Lion`:
+```graphql
+extend type Query @fieldPolicy(forField: "allAnimals", keyArgs: "habitat species")
+```
+
+#### List Arguments
+
+The `@fieldPolicy` directive does support fields which return a list. There are a few caveats with the usage of lists:
+
+- If a field has a list return type, it must contain exactly one list input parameter.
+- Conversely if a list contains a list input parameter, it must return a list.
+- Fields cannot have more than one list input parameter.
```

Comment:
Should this be "List fields cannot..."?

Reply:
It's also not really necessary since the first point states "...must contain exactly one list input parameter".